src/server/game/Spells/Spell.cpp

```diff
@@ -4973,6 +4973,23 @@ SpellCastResult Spell::CheckCast(bool strict)
         if (castResult != SPELL_CAST_OK)
             return castResult;
 
+        // If it's not a melee spell, check if vision is obscured by SPELL_AURA_INTERFERE_TARGETTING
+        if (m_spellInfo->DmgClass != SPELL_DAMAGE_CLASS_MELEE)
```

Comment:
Be careful, this condition should belong only to Camouflage (try check SPELL_AURA_MOD_CAMOUFLAGE), not for Smoke Bomb.
Anyway I am no sure if SPELL_DAMAGE_CLASS_MELEE is good check, according to commnets some melee spells do not work, some does.